plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx

```diff
@@ -77,6 +77,13 @@ export const UserSettingsFeatureFlags = () => {
         ...prevState,
         [flagName]: newState === FeatureFlagState.Active,
       }));
+
+      const updatedFlags = sortFlags(
+        featureFlagsApi.getRegisteredFlags(),
+        featureFlagsApi,
+      );
+
+      setFeatureFlags(updatedFlags);
```

Comment:
Skipping this step is actually by design, we don't want the flags to re-order while making changes as it's quite jarring. The flag value still gets updated under the hood though, the `featureFlags` state here only represents the ordering of the flags, not whether they're active or not, that's the `featureFlagsApi.save()` call above